src/AzureAuth.Test/PublicClientAuthTest.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace AzureAuth.Test
+{
+    using System;
+    using System.Collections.Generic;
+
+    using FluentAssertions;
+
+    using Microsoft.Authentication.AzureAuth;
+    using Microsoft.Authentication.MSALWrapper;
+    using Microsoft.Authentication.TestHelper;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.IdentityModel.JsonWebTokens;
+    using Microsoft.Office.Lasso.Interfaces;
+    using Microsoft.Office.Lasso.Telemetry;
+
+    using Moq;
+
+    using NLog.Targets;
+
+    using NUnit.Framework;
+
+    internal class PublicClientAuthTest
+    {
+        private readonly Guid client = Fake.Client;
+        private readonly Guid tenant = Fake.Tenant;
+        private readonly IEnumerable<string> scopes = Fake.Scopes;
+        private readonly string domain = Fake.Domain;
+        private readonly string prompt = "AzuerAuth.Test";
+        private readonly TimeSpan timeout = TimeSpan.FromSeconds(10);
+
+        private ILogger logger;
+        private MemoryTarget logTarget;
+        private Mock<IEnv> env;
+        private Mock<ITelemetryService> telemetryService;
+        private Mock<IMsalWrapper> tokenFetcher;
+
+        [SetUp]
+        public void SetUp()
+        {
+            (this.logger, this.logTarget) = MemoryLogger.Create();
+            this.env = new Mock<IEnv>(MockBehavior.Strict);
+            this.telemetryService = new Mock<ITelemetryService>(MockBehavior.Strict);
+            this.tokenFetcher = new Mock<IMsalWrapper>(MockBehavior.Strict);
+        }
+
+        public PublicClientAuth Subject() => new PublicClientAuth(this.logger, this.env.Object, this.telemetryService.Object, this.tokenFetcher.Object);
+
+        [Test]
+        public void Contructor_Works()
+        {
+            this.Subject().Should().NotBeNull();
+        }
+
+        [Test]
+        public void Constructor_No_Nulls_Allowed()
+        {
+            Action nullLogger = () => new PublicClientAuth(null, null, null, null);
+            nullLogger.Should().Throw<ArgumentNullException>().WithParameterName("logger");
+
+            Action nullEnv = () => new PublicClientAuth(this.logger, null, null, null);
+            nullEnv.Should().Throw<ArgumentNullException>().WithParameterName("env");
+
+            Action nullTelemetryService = () => new PublicClientAuth(this.logger, this.env.Object, null, null);
+            nullTelemetryService.Should().Throw<ArgumentNullException>().WithParameterName("telemetryService");
+
+            Action nullTokenFetcher = () => new PublicClientAuth(this.logger, this.env.Object, this.telemetryService.Object, null);
+            nullTokenFetcher.Should().Throw<ArgumentNullException>().WithParameterName("tokenFetcher");
+        }
+
+        [Test]
+        public void Good_Auth_Returns_TokenResult()

```

Comment:
I would like to see some acknowledgement that this is not a typical unit test, but really more like an integration test.